core/src/Instance/B2CAuthority.cs

```diff
@@ -37,37 +38,85 @@ internal class B2CAuthority : AadAuthority
     {
         public const string Prefix = "tfp"; // The http path of B2C authority looks like "/tfp/<your_tenant_name>/..."
         public const string B2CCanonicalAuthorityTemplate = "https://{0}/{1}/{2}/{3}/";
+        public const string MicrosoftOnline = "https://login.microsoftonline.com";
+        public const string OpenIdConfigurationEndpoint = "v2.0/.well-known/openid-configuration";
+        internal const string B2CTrustedHost = "b2clogin.com";
 
-        internal B2CAuthority(string authority, bool validateAuthority) 
+        internal B2CAuthority(string authority, bool validateAuthority)
             : base(authority, validateAuthority)
         {
+            AuthorityType = AuthorityType.B2C;
+            ValidateAuthority = validateAuthority;
+
             Uri authorityUri = new Uri(authority);
-            string[] pathSegments = authorityUri.AbsolutePath.Substring(1).Split(new [] { '/'}, StringSplitOptions.RemoveEmptyEntries);
+            string[] pathSegments = authorityUri.AbsolutePath.Substring(1).Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
             if (pathSegments.Length < 3)
             {
                 throw new ArgumentException(CoreErrorMessages.B2cAuthorityUriInvalidPath);
             }
 
             CanonicalAuthority = string.Format(CultureInfo.InvariantCulture, B2CCanonicalAuthorityTemplate, authorityUri.Authority,
                 pathSegments[0], pathSegments[1], pathSegments[2]);
-
-            AuthorityType = AuthorityType.B2C;
         }
 
-        protected override async Task<string> GetOpenIdConfigurationEndpointAsync(
-            IHttpManager httpManager, 
-            ITelemetryManager telemetryManager, 
-            string userPrincipalName, 
+        internal override async Task UpdateCanonicalAuthorityAsync(
+            IHttpManager httpManager,
+            ITelemetryManager telemetryManager,
             RequestContext requestContext)
         {
-            if (ValidateAuthority && !IsInTrustedHostList(new Uri(CanonicalAuthority).Host))
+            if (ValidateAuthority && !InTrustedHostList(new Uri(CanonicalAuthority).Host))
             {
                 throw new ArgumentException(CoreErrorMessages.UnsupportedAuthorityValidation);
             }
+            
+            var metadata = await AadInstanceDiscovery
```

Comment:
Doesn't calling base.UpdateCannonicalAuthorityAsync do the same thing?

Reply:
we have to change the canonical authority to get the metadata back because this is not supported in b2c yet. 

---
In reply to: [230919572](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1383#discussion_r230919572) [](ancestors = 230919572)